### ISSUE TITLE
Add support for ppc64le arch in kube-build image

### ISSUE THREAD

**Issue by qq511700505** [kubernetes/kubernetes#19185]
we are trying build kubernetes binaries with docker on ppc64le . But the Dockerfile in /build/build-image/ gets etcd binary by downloading 'etcd-$ETCD_VERSION-linux-amd64.tar.gz' (Line 37-43)

we consider  it will be better if change it into getting the source code of etcd and building binary according to  current architecture.

**Comment by luxas** [comment]
@qq511700505 @fgrzadkowski  I'm working on this right now... Will send a PR soon

Related: #17981

**Comment by qq511700505** [comment]
Now the main problem is that I cann't get golang docker images for ppc64le. [ps:To bulid golang binaries we need a go binary first.] But there is no official download link for golang binary for ppc64le, which is the key problem, I think.
@luxas  @fgrzadkowski

**Comment by luxas** [comment]
Well, you have to find a `debian`-ish `ppc64le` base image or build your own. Then you could download [kubernetes-on-arm](https://github.com/luxas/kubernetes-on-arm), and see how `luxas/go` and `kubernetesonarm/build` are built. There are instructions for how to build a) a `golang` image from scratch b) kubernetes form source

But in this case (as I'm working on it) I think we should focus on cross-compiling from `amd64`. As you see, #19216 is out to fix that, for `etcd`. 
I have a other PRs coming for e.g. `flannel`, `hyperkube`, `kube2sky` and `skydns` that does the same thing.

**Comment by seelam** [comment]
Here is the docker image with golang for ppc64le: https://hub.docker.com/r/ppc64le/golang/

**Comment by luxas** [comment]
@qq511700505 Now is the code merged. So you can do this (in the future)
```
cd cluster/images/etcd
sudo make build ARCH=ppc64le BASEIMAGE=ppc64le/busybox TAG=2.3.0
```
As I said, `etcd` doesn't compile on `2.2.x` and `2.3.0` isn't released for the moment. But, in the future you may do that. This commit fixed the issue, but it isn't in any release yet: https://github.com/coreos/etcd/commit/b4e0dbe721637bb6b36575a06f6f05b9251946f1

So you have to wait... unfortunately. 
Or you could edit the script and compile from master, but that would result in a totally unofficial binary

**Comment by luxas** [comment]
@qq511700505 Now you can build your own `etcd` image for Kubernetes (on an amd64 host):
```bash
cd cluster/images/etcd
sudo make build ARCH=ppc64le BASEIMAGE=ppc64le\\/busybox TAG=2.2.5
```

Soon #19769 will land and then Kubernetes will release binaries for `ppc64le` automatically on every release. So when #19769 is merged, this issue may be closed

**Comment by luxas** [comment]
@qq511700505 @pensu Kubernetes on ppc64le experimental release will be available when `v1.3.0-alpha.3` is released. It's only one command away with the docker guide.

I see this as fixed. Reopen if you think there are things left to do.
